docs/src/main/asciidoc/grpc-getting-started.adoc

```diff
@@ -0,0 +1,318 @@
+}
+
+// The request message containing the user's name.
+message HelloRequest {
+    string name = 1;
+}
+
+// The response message containing the greetings
+message HelloReply {
+    string message = 1;
+}
+----
+
+This `proto` file defines a simple service interface with a single method (`SayHello`), and the exchanged messages (`HelloRequest` containing the name and `HelloReply` containing the greeting message).
+
+Before coding, we need to generate the classes used to implement and consume gRPC services.
+In a terminal, run:
+
+[source, bash]
+----
+$ mvn compile
+----
+
+Once generated, you can look at the `target/generated-sources/protobuf` directory:
+
+[source, txt]
+----
+target/generated-sources/protobuf
+├── grpc-java
+│   └── io
+│       └── quarkus
+│           └── example
+│               └── GreeterGrpc.java
+└── java
+    └── io
+        └── quarkus
+            └── example
+                ├── HelloReply.java
+                ├── HelloReplyOrBuilder.java
+                ├── HelloRequest.java
+                ├── HelloRequestOrBuilder.java
+                ├── HelloWorldProto.java
+                └── MutinyGreeterGrpc.java
+----
+
+These are the classes we are going to use.
+
+IMPORTANT: Every time you update the `proto` files, you need to re-generate the classes (using `mvn compile`).
+
+== Implementing a gRPC service
+
+Now that we have the generated classes let's implement our _hello_ service.
+
+With Quarkus gRPC, implementing a service requires to _extend_ the generated service base implementation and exposes it as a `@Singleton` CDI bean.
+
+IMPORTANT: Don't use `@ApplicationScoped` as the gRPC service implementation cannot be proxied.
+
+=== Implementing a service
+
+Create the `src/main/java/org/acme/HelloService.java` file with the following content:
+
+[source, java]
+----
+package org.acme;
+
+import io.grpc.stub.StreamObserver;
+import io.quarkus.example.GreeterGrpc;
+import io.quarkus.example.HelloReply;
+import io.quarkus.example.HelloRequest;
+
+import javax.inject.Singleton;
+
+@Singleton                                                                                    // <1>
+public class HelloService extends GreeterGrpc.GreeterImplBase {                               // <2>
+
+    @Override
+    public void sayHello(HelloRequest request, StreamObserver<HelloReply> responseObserver) { // <3>
+        String name = request.getName();
+        String message = "Hello " + name;
+        responseObserver.onNext(HelloReply.newBuilder().setMessage(message).build());         // <4>
+        responseObserver.onCompleted();                                                       // <5>
+    }
+}
+
+----
+1. Expose your implementation as bean
+2. Extends the `ImplBase` class. This is a generated class.
+3. Implement the methods defined in the service definition (here we have a single method)
+4. Build and send the response
+5. Close the response
+
+Quarkus also provides an additional model with Mutiny.
```

Comment:
maybe add what Mutiny is in 3 words?

Reply:
Fixed. I also linked the getting-started-with-reactive guide.